config/config.go

```diff
@@ -265,15 +265,27 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		}
 		jobNames[scfg.JobName] = struct{}{}
 	}
+	rwNames := map[string]struct{}{}
 	for _, rwcfg := range c.RemoteWriteConfigs {
 		if rwcfg == nil {
 			return errors.New("empty or null remote write config section")
 		}
+		// Skip empty names, we fill their name with their config hash in remote write code.
+		if _, ok := rwNames[rwcfg.Name]; ok && rwcfg.Name != "" {
```

Comment:
What would you think of setting the default name while parsing? Then the remote code could depend on there always being a name.

Might make testing or re-use tricky since a name would always have to be present though...

Reply:
That would be fine imo.

Reply:
This would require the `toHash` function to be available in `config/config.go` by imports or declaring the function in `config`.

Reply:
Good point. Lets keep the defaulting logic in the remote code.